### PR TITLE
Brokered Hybrid Spa

### LINKED ISSUE #3994: [Feature Request] Support "spa_accountId" response parameter from STS
**Is your feature request related to a problem? Please describe.**
Azure Portal is working on a feature (called the WebNativeBridge) which enables it to use OS single sign-in state. This feature initializes MSAL.JS with a parameter "accountId" which is a handle into the native operation system instead of the normal OAuth authorization code parameter. When Azure Portal uses this feature in its "hybrid SPA" implementation, the server returns this parameter to the confidential client in the field "spa_accountId" instead of the existing parameter "spa_code". MSAL.JS needs to use this parameter to initialize itself.

**Describe the solution you'd like**
Like AuthResult.SPA_CODE, expose AuthResult.SPA_ACCOUNTID. 

**Describe alternatives you've considered**
Alternatively, expose AuthResult.AdditionalParameters, so we can access AuthResult.AdditionalParameters["spa_accountId"].

Another alternative considered - putting spa code and spa accountid in the same field - does not work. Both codes and accountids are opaque. MSAL.JS needs to know whether the opaque value used to initialize it is an authorization code or an accountId.

**Additional context**
Part of zod. 

** Algorithm **

image

Note ^^  that MSAL.NET does not make /authorization calls, it just formats the URL and using `WithExtraQueryParameters` is sufficient here.

### PR DESCRIPTION
**Pull request by bgavrilMS**
Fixes #3994 

**Changes proposed in this request**
See Item  #3994  for scenario

**Testing**

unit

**Performance impact** 
minor, just an extra field to parse in the /token response

**Documentation**
- have not updated Hybrid SPA scenario

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/AuthenticationResult.cs [hunk 1]
```diff
@@ -39,7 +39,8 @@ public partial class AuthenticationResult
         /// <param name="tokenType">The token type, defaults to Bearer. Note: this property is experimental and may change in future versions of the library.</param>
         /// <param name="authenticationResultMetadata">Contains metadata related to the Authentication Result.</param>
         /// <param name="claimsPrincipal">Claims from the ID token</param>
-        /// <param name="spaAuthCode">Auth Code returned by the Microsoft identity platform when you use AcquireTokenByAuthorizeCode.WithSpaAuthorizationCode(). This auth code is meant to be redeemed by the frontend code.</param>
+        /// <param name="spaAuthCode">Auth Code returned by the Microsoft identity platform when you use AcquireTokenByAuthorizeCode.WithSpaAuthorizationCode(). This auth code is meant to be redeemed by the frontend code. See https://aka.ms/msal-net/spa-auth-code</param>
```
Comment (gladjohn) [inline src/client/Microsoft.Identity.Client/AuthenticationResult.cs:42]:
```suggestion
        /// <param name="spaAuthCode">Auth Code returned by the Microsoft identity platform when you use AcquireTokenByAuthorizationCode.WithSpaAuthorizationCode(). This auth code is meant to be redeemed by the frontend code. See https://aka.ms/msal-net/spa-auth-code</param>
```

File: src/client/Microsoft.Identity.Client/AuthenticationResult.cs [hunk 2]
```diff
@@ -268,6 +273,17 @@ public partial class AuthenticationResult
...
         public string SpaAuthCode { get; }
 
+        /// <summary>
+        /// Get the Accound ID of a user, instead of <see cref="SpaAuthCode"/>, if the user's account
+        /// is connected to Windows. 
+        /// 
+        /// See https://aka.ms/msal-net/spa-auth-code#advanced-scenario---bridged-spa-auth-code-protocol for details.
+        /// </summary>
+        /// <remarks>
+        /// To opt-in this feature, add `nativebroker=1` to the authorization url.
```
Comment (gladjohn) [inline src/client/Microsoft.Identity.Client/AuthenticationResult.cs:283]:
```suggestion
        /// To opt-in to this feature, add `nativebroker=1` to the authorization url. 
```

**Review by gladjohn — APPROVED** [review]
:shipit:

File: tests/Microsoft.Identity.Test.Unit/PublicApiTests/AuthenticationResultTests.cs [hunk 3]
```diff
@@ -131,5 +130,69 @@ public void AuthenticationResult_PublicApi()
...
+                // Strongly typed properties should not be exposed (we don't want to expose refresh token)
+                Assert.IsFalse(extMap.ContainsKey("scope"));
+                Assert.IsFalse(extMap.ContainsKey("expires_in"));
+                Assert.IsFalse(extMap.ContainsKey("ext_expires_in"));
+                Assert.IsFalse(extMap.ContainsKey("access_token"));
+                Assert.IsFalse(extMap.ContainsKey("refresh_token"));
+                Assert.IsFalse(extMap.ContainsKey("id_token"));
+                Assert.IsFalse(extMap.ContainsKey("client_info"));
+
+                // only scalar properties shouldbe in the map
```
Comment (neha-bhargava) [inline tests/Microsoft.Identity.Test.Unit/PublicApiTests/AuthenticationResultTests.cs:183]: Nit
```suggestion
                // only scalar properties should be in the map
```

File: tests/Microsoft.Identity.Test.Unit/CoreTests/OAuth2Tests/TokenResponseTests.cs [hunk 4]
```diff
@@ -2,7 +2,9 @@
 // Licensed under the MIT License.
 
 using System;
+using System.Collections.Generic;
```
Comment (neha-bhargava) [inline tests/Microsoft.Identity.Test.Unit/CoreTests/OAuth2Tests/TokenResponseTests.cs:5]: Nit: not needed.

**Review by pmaytak — APPROVED** [review]

**Pull request merged**